modules/compute/vm-instance/variables.tf

```diff
@@ -136,6 +136,73 @@ variable "subnetwork_self_link" {
   default     = null
 }
 
+variable "network_interfaces" {
+  description = <<-EOT
+    A list of network interfaces. The options match that of the terraform
+    network_interface block of google_compute_instance. For descriptions of the
+    subfields or more information see the documentation:
+    https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#nested_network_interface
+
+    **_NOTE:_** If `network_interfaces` are set, `network_self_link` and
+    `subnetwork_self_link` will be ignored, even if they are provided through
+    the `use` field. `bandwidth_tier` and `disable_public_ips` also do not apply
+    to network interfaces defined in this variable.
+
+    Subfields:
+    network            (string, required if subnetwork is not supplied)
+    subnetwork         (string, required if network is not supplied)
+    subnetwork_project (string, optional)
+    network_ip         (string, optional)
+    nic_type           (string, optional, choose from ["GVNIC", "VIRTIO_NET"])
+    stack_type         (string, optional, choose from ["IPV4_ONLY", "IPV4_IPV6"])
+    queue_count        (number, optional)
+    access_config      (object, optional)
+    ipv6_access_config (object, optional)
+    alias_ip_range     (list(object), optional)
+    EOT
+  type = list(object({
+    network            = string,
+    subnetwork         = string,
+    subnetwork_project = string,
+    network_ip         = string,
+    nic_type           = string,
+    stack_type         = string,
+    queue_count        = number,
+    access_config = list(object({
+      nat_ip                 = string
+      public_ptr_domain_name = string
+      network_tier           = string
+    })),
+    ipv6_access_config = list(object({
+      public_ptr_domain_name = string,
+      network_tier           = string
+    })),
+    alias_ip_range = list(object({
+      ip_cidr_range         = string
+      subnetwork_range_name = string
+    }))
+  }))
+  default = []
+  validation {
+    condition = alltrue([
+      for ni in var.network_interfaces : (ni.network == null) != (ni.subnetwork == null)
+    ])
+    error_message = "All additional network interfaces must define exactly one of \"network\" or \"subnetwork\"."
+  }
+  validation {
+    condition = alltrue([
+      for ni in var.network_interfaces : ni.nic_type == "GVNIC" || ni.nic_type == "VIRTIO_NET" || ni.nic_type == null
```

Comment:
nit: above 3 values, I'd start considering [contains](https://www.terraform.io/language/functions/contains)

Reply:
I like the idea, but `contains` doesn't seem to play well with values that can be null:
```
│ Error: Invalid function argument
│ 
│   on modules/vm-instance/variables.tf line 200, in variable "network_interfaces":
│  200:       for ni in var.network_interfaces : ni.stack_type == null || contains(["IPV4_ONLY", "IPV4_IPV6"], ni.stack_type)
│     ├────────────────
│     │ ni.stack_type is null
│ 
│ Invalid value for "value" parameter: argument must not be null.
```

So the options seem to be:
1. keep it as it is
1. require these values to be specified every time